frontend/contact.component.ts

```diff
@@ -1,28 +1,53 @@
-import { Component }      from "@angular/core";
-import { Router }         from "@angular/router";
+import { Component, OnInit, NgZone, OnDestroy }  from "@angular/core";
+import { Router }             from "@angular/router";
 
-import { ContactService } from "./contact.service";
-import { Contact }        from "./contact";
+import { ContactService }     from "./contact.service";
+import { Contact }            from "./contact";
+
+import { CaptchaService }     from "./captcha.service";
+import { ConfigurationService }      from "./config.service";
 
 import { TOOLTIP_DIRECTIVES } from "ng2-bootstrap/components/tooltip";
 
 @Component({
     selector: "sg-contact",
     templateUrl: "contact.component.html",
-    providers: [ContactService],
+    providers: [ContactService, CaptchaService, ConfigurationService],
     directives: [TOOLTIP_DIRECTIVES],
 })
-export class ContactComponent {
+export class ContactComponent implements OnInit, OnDestroy {
     private contactErrorMessage: string;
     private contactSuccessMessage: string;
     private contact: Contact;
     private uploads: FileList;
     private maxFileNameLength = 50;
-    private filesValid = true;
+    private filesValid;
+    private captchaVerfied;
+    private script;
+    private grecaptchaKey;
 
-    constructor( private contactService: ContactService, private router: Router) {
+    constructor( private contactService: ContactService, private captchaService: CaptchaService,
+                 private configService: ConfigurationService, private router: Router, private zone: NgZone) {
         this.contact = new Contact("", "", "", "", null, null);
         this.filesValid = true;
+        this.captchaVerfied = false;
+        window["verifyCallback"] = this.verifyCallback.bind(this);
+        this.grecaptchaKey = this.configService.getConfigEntry("data-sitekey");
+    }
+
+    public ngOnInit() {
+        // Add script tag manually as it does not work from frontend.html, g-recaptcha not displayed
+        let doc = <HTMLDivElement> document.body;
+        this.script = document.createElement("script");
+        this.script.innerHTML = "";
+        this.script.src = "https://www.google.com/recaptcha/api.js";
+        this.script.async = true;
+        this.script.defer = true;
+        doc.appendChild(this.script);
+    }
```

Comment:
Theoretisch sollte es reichen, dass ein script tag in der frontend.html landet, ähnlich der maps.googleapis.com .
Problem, egal wie und an welcher erkennt er es nicht und rendert dann das Captcha nicht. Schien generell ein Problem mit einer Subkomponente zu sein, die nicht direkt in die index(oder bei uns frontend.html) eingebunden ist, da es auch in einigen Angular2 Packages die das ganze in eine Component packen, die man verwenden könnte, nicht anders gemacht wird.
Hätte ich auch besser gefunden, einfach nur das Script Tag drinnen zu haben, aber leider nicht funktioniert.